File: glucare_java/glucare/src/main/java/com/uade/glucare/model/FoodItem.java
package com.uade.glucare.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Column;

@Entity
@Table(name = "food_item")
public class FoodItem {

//     Atributos:
// nombre: String
// cantidadCarbohidratos: double
// cantidadProteinas: double
// cantidadGrasas: double

    @Id
    @GeneratedValue
    @Column(name = "id")
    private Long id;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "descripcion")
    private String descripcion;

    @Column(name = "cantidad_calorias")
    private double cantidadCalorias;

    @Column(name = "cantidad_carbohidratos")
    private double cantidadCarbohidratos;

    @Column(name = "cantidad_fibra")
    private double cantidadFibra;

    @Column(name = "cantidad_proteinas")
    private double cantidadProteinas;

    @Column(name = "cantidad_grasas")
    private double cantidadGrasas;

    @Column(name = "vitamina")
    private String vitamina;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;


    public Long getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public double getCantidadCalorias() {
        return cantidadCalorias;
    }

    public double getCantidadCarbohidratos() {
        return cantidadCarbohidratos;
    }

    public double getCantidadFibra() {
        return cantidadFibra;
    }

    public double getCantidadProteinas() {
        return cantidadProteinas;
    }
 
    public String getVitamina() {
        return vitamina;
    }

    public double getCantidadGrasas() {
        return cantidadGrasas;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public void setCantidadCalorias(double cantidadCalorias) {
        this.cantidadCalorias = cantidadCalorias;
    }

    public void setCantidadFibra(double cantidadFibra) {
        this.cantidadFibra = cantidadFibra;
    }


    public void setCantidadCarbohidratos(double cantidadCarbohidratos) {
        this.cantidadCarbohidratos = cantidadCarbohidratos;
    }

    public void setCantidadProteinas(double cantidadProteinas) {
        this.cantidadProteinas = cantidadProteinas;
    }

    public void setCantidadGrasas(double cantidadGrasas) {
        this.cantidadGrasas = cantidadGrasas;
    }

    public void setVitamina(String vitamina) {
        this.vitamina = vitamina;
    }
    
}

File: glucare_flutter/lib/model/FoodItem.dart
import 'package:glucare/model/User.dart';

class FoodItem {
  final int id;
  final String nombre;
  final double cantidadCarbohidratos;
  final double cantidadProteinas;
  final double cantidadGrasas;
  final User user;

  FoodItem({
    required this.id,
    required this.nombre,
    required this.cantidadCarbohidratos,
    required this.cantidadProteinas,
    required this.cantidadGrasas,
    required this.user,
  });

  factory FoodItem.fromJson(Map<String, dynamic> json) {
    return FoodItem(
      id: json['id'],
      nombre: json['nombre'],
      cantidadCarbohidratos: json['cantidadCarbohidratos'],
      cantidadProteinas: json['cantidadProteinas'],
      cantidadGrasas: json['cantidadGrasas'],
      user: User.fromJson(json['user']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'cantidadCarbohidratos': cantidadCarbohidratos,
      'cantidadProteinas': cantidadProteinas,
      'cantidadGrasas': cantidadGrasas,
      'user': user.toJson(),
    };
  }
}